app/src/main/java/com/ayush/weatherapp/customViews/WrapContentHeightViewPager.java

```diff
@@ -0,0 +1,33 @@
+package com.ayush.weatherapp.customViews;
+
+import android.content.Context;
+import android.support.v4.view.ViewPager;
+import android.util.AttributeSet;
+import android.view.View;
+
+//Reference: https://stackoverflow.com/questions/8394681/android-i-am-unable-to-have-viewpager-wrap-content
+public class WrapContentHeightViewPager extends ViewPager {
```

Comment:
Is this absolutely necessary?

Reply:
yes. to make the view pager wrap content or else it is always match parent by default.